torch/csrc/autograd/python_function.cpp

```diff
@@ -571,53 +565,37 @@ static void _trace_create(PyObject* op_obj, THPFunction* bw_obj,
     }
   }
 
-  auto state_lock = tracing_state->lock();
-
-  // Note [getValueTrace can allocate nodes]
-  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-  // When an input variable is not traced, we create a constant instruction
-  // to represent it.  This means that you must invoke getValueTrace() BEFORE
-  // actually constructing the function that takes these variables as inputs.
-  // If we do it the other order, the graph will be in the wrong topological
-  // order.
-
-  // See Note [getValueTrace can allocate nodes]
-  std::vector<Value*> value_traces;
-  std::vector<VariableFlags> var_flags;
-  value_traces.reserve(input_vars.size());
-  for (auto& i : input_vars) {
-    value_traces.emplace_back(tracer::getValueTrace(tracing_state, i));
-    var_flags.push_back(VariableFlags::of(i));
-  }
+  Py_INCREF(op_obj);
```

Comment:
Do you want me to double check refcounting changes?

Reply:
I'm pretty confident about this one. It corresponds to the `THPObjectPtr()` construction below.

We could add a "retain" argument to the THPObjectPtr constructor like we do in `at::Tensor` to avoid the separate incref call.